lib/measures/ConstrainSupplyAirTemperatureResetVerification/measure.py

```diff
@@ -0,0 +1,270 @@
+            },
+        }
+
+        return workflow
+
+    def get_verification_case(
+        self,
+        output_dataset_path,
+        supply_outlet_node,
+        air_loop,
+        design_zone_cooling_air_temp,
+    ):
+        supply_air_temperature_reset_verification_case = {
+            "no": 1,
+            "run_simulation": False,
+            "expected_result": "pass",
+            "simulation_IO": {"output": output_dataset_path},
+            "datapoints_source": {
+                "idf_output_variables": {
+                    "T_sa_set": {
+                        "subject": supply_outlet_node.nameString(),
+                        "variable": f"{air_loop.nameString()} Supply Outlet Temperature",
+                        "frequency": "detailed",
+                    }
+                },
+                "parameters": {"T_z_coo": design_zone_cooling_air_temp},
+            },
+            "verification_class": "SupplyAirTempReset",
+        }
+
+        return {"cases": [supply_air_temperature_reset_verification_case]}
+
+    def run(
+        self,
+        model: openstudio.model.Model,
+        runner: openstudio.measure.OSRunner,
+        user_arguments: openstudio.measure.OSArgumentMap,
+    ):
+        """
+        Define what happens when the measure is run
+        """
+        super().run(model, runner, user_arguments)
+
+        if not (runner.validateUserArguments(self.arguments(model), user_arguments)):
+            return False
+
+        air_loop_name = runner.getStringArgumentValue("air_loop_name", user_arguments)
+        design_zone_cooling_air_temp = runner.getDoubleArgumentValue(
+            "design_zone_cooling_air_temp", user_arguments
+        )
+        output_dataset_path = runner.getStringArgumentValue(
+            "output_dataset_path", user_arguments
+        )
+        output_dir = runner.getStringArgumentValue("output_dir", user_arguments)
+
+        runner.registerInitialCondition("Init")
+
+        air_loop = model.getAirLoopHVACByName(air_loop_name)
+        if not air_loop.get():
+            runner.registerError(
+                f"AirLoopHVAC '{air_loop_name}' not found in the model"
+            )
+            return False
+
+        air_loop = air_loop.get()
+
+        supply_outlet_node = air_loop.supplyOutletNode()
+
+        output_variable = openstudio.model.OutputVariable(
+            "System Node Temperature", model
+        )
+        output_variable.setName(f"{air_loop.name().get()} Supply Outlet Temperature")
+        output_variable.setKeyValue(f"{supply_outlet_node.name().get()}")
+
+        runner.registerInfo("Added OutputVariable for supply outlet node temperature")
+
+        verification_cases = self.get_verification_case(
+            output_dataset_path,
+            supply_outlet_node,
+            air_loop,
+            design_zone_cooling_air_temp,
+        )
+        with open(f"{output_dir}/supply_air_temperature_reset_verification_case.json", "w") as f:
```

Comment:
Does the supply_air_temperature_reset_verification_case.json exists all the time?
If not, suggest adding a file exist check first.

Reply:
I added line 261 to handle this. open's "w" mode will create a new file if it doesn't exist, but it won't create any directories.